public/search.js

```diff
@@ -1,52 +1,213 @@
+  if (getSelected('valDropdown') !== 'None') {
+    $('#searchQueryButton, #dragUp, #dragDown, .valTimesDisplay').css({'visibility': 'visible'});
   }
-);
+});
 
-//function when the next button is clicked
-$('#next').click(function()
-  {
-    firstDoc+=10;
-    lastDoc+=10;
-    displayDocs(firstDoc, lastDoc);
+$('#sysDropdown').change(function() {
+  $('#searchQueryButton, #dragRight, #dragLeft, .sysTimesDisplay').css({'visibility': 'hidden'});
+  if (getSelected('sysDropdown') !== 'None') {
+    $('#searchQueryButton, #dragRight, #dragLeft, .sysTimesDisplay').css({'visibility': 'visible'});
   }
-);
+});
 
-//function when the prev button is clicked
-$('#prev').click(function()
-  {
-    firstDoc-=10;
-    lastDoc-=10;
-    displayDocs(firstDoc, lastDoc);
+$('#searchQueryButton').click(function() {
+  runSearchQuery();
+});
+
+$('#resetButton').click(function() {
+  window.location.reload();
+});
+
+function runSearchQuery() {
+  var selectedColl = getSelected('dropdown');
+  var valSelectedOp = getSelected('valDropdown');
+  var sysSelectedOp = getSelected('sysDropdown');
+
+  var valAxis = '';
+  var sysAxis = '';
+
+  var valStart = '';
+  var valEnd = '';
+
+  if(valSelectedOp !== 'None') {
+    valAxis = 'myValid';
+    valStart = document.getElementById('startValBox').value;
+    valEnd = document.getElementById('endValBox').value;
+    if (valStart >= valEnd) {
+      alert('Valid start time cannot be greater than or equal to valid end time');
+      return;
+    }
+    valStart = new Date(valStart).toISOString();
+    valEnd = new Date(valEnd).toISOString();
+  }
+
+  var sysStart = '';
+  var sysEnd = '';
+
+  if(sysSelectedOp !== 'None') {
+    sysAxis = 'mySystem';
+    sysStart = document.getElementById('startSysBox').value;
+    sysEnd = document.getElementById('endSysBox').value;
+    if (sysStart >= sysEnd) {
+      alert('System start time cannot be greater than or equal to system end time');
+      return;
+    }
+    sysStart = new Date(sysStart).toISOString();
+    sysEnd = new Date(sysEnd).toISOString();
+  }
+
+  if(valSelectedOp === 'None' && sysSelectedOp === 'None' ) {
+    ajaxTimesCall(selectedColl, null);
+    return;
+  }
+
+  $.ajax(
+    {
+      url: '/v1/resources/operators?rs:collection='+selectedColl+'&rs:valAxis='+valAxis+'&rs:valSelectedOp='+valSelectedOp+'&rs:sysAxis='+sysAxis+'&rs:sysSelectedOp='+sysSelectedOp+'&rs:valStart='+valStart+'&rs:valEnd='+valEnd+'&rs:sysStart='+sysStart+'&rs:sysEnd='+sysEnd,
+      success: function(response, textStatus)
+      {
+        displayQuery(response);
+        ajaxTimesCall(response.collection, response);
+      },
+      error: function(jqXHR, textStatus, errorThrown)
+      {
+        console.log('problem');
+      }
+    }
+  );
+}
+
+function displayQuery(response) {
+  document.getElementById('queryText').innerHTML = response.query;
+}
+
+function createCorrData(response) {
```

Comment:
What does "Corr" mean? Please either expand (preferred if not crazy long) or add a comment